### PR TITLE
use of TabTabGo and Add interface ISaveNotificatoin to save Notificaton to storage

### PR DESCRIPTION
**Pull request by Mohammad-Sarya-Tulimat**
ISaveWebStreamMessage:  it have same logic of  IPushEvent implementation of storage  
but it is other interface used with storage only so we can use it when we don't need to push all Messages to storage

### CODE REVIEW

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/EfCoreStorageBuilder.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/EfCoreStorageBuilder.cs:1]: Remove Tabtabgo as prefix and add schema as option

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Mappers/UserNotificationMapper.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Mappers/UserNotificationMapper.cs:1]: Remove TabTabGo as prefix table name and add option for schema

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Mappers/NotificationMapper.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Mappers/NotificationMapper.cs:1]: Remove TabTabGo as prefix table name and add option for schema

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Module/PageingResult.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Module/PageingResult.cs:1]: Use the standard PageList and PageOptions from TabTabGo.Core

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Module/PagingParameters.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Module/PagingParameters.cs:1]: Use TabTabGo.Core PageOptions

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/QueryExt.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/QueryExt.cs:1]: This nice to move to TabTabGo.Core

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationRepository.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationRepository.cs:1]: rename UserId to userId

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/INotificationServices.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/INotificationServices.cs:1]: Use PageList and PageOption

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageService.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageService.cs:1]: Still don't see a need for it

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageSucessOnDecorator.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageSucessOnDecorator.cs:1]: Still don't see a need

File: server/src/TabTabGo.WebStream/TabTabGo.WebStream/Model/WebStreamMessage.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/TabTabGo.WebStream/Model/WebStreamMessage.cs:1]: Don't see a need for it

File: server/src/TabTabGo.WebStream/TabTabGo.WebStream/TabTabGo.WebStream.csproj
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/TabTabGo.WebStream/TabTabGo.WebStream.csproj:1]: Convert to .net8

**Review by gmsamara — CHANGES_REQUESTED** [review]
Few points
1- Use PageList and PageOption instead of PageResultBuilder and Page Paramters
2- I don't see a need to store Messages to database
3- Nee to add logic store messages to log

File: server/src/TabTabGo.WebStream/TabTabGo.WebStream/Model/WebStreamMessage.cs
Comment (Mohammad-Sarya-Tulimat) [reply server/src/TabTabGo.WebStream/TabTabGo.WebStream/Model/WebStreamMessage.cs:1]: how ??

Comment (Mohammad-Sarya-Tulimat) [reply server/src/TabTabGo.WebStream/TabTabGo.WebStream/Model/WebStreamMessage.cs:1]: it is the message object we pass in and out

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageService.cs
Comment (Mohammad-Sarya-Tulimat) [reply server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageService.cs:1]: simply, don't add it in configuration if you don't need it

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageSucessOnDecorator.cs
Comment (Mohammad-Sarya-Tulimat) [reply server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Services/PushToStorageSucessOnDecorator.cs:1]: simply, don't add it in configuration if you don't need it

**Review by gmsamara — APPROVED** [review]

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationUserRepository.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationUserRepository.cs:1]: Why we need these function use what in GenericRepoository
 
     Task<List<NotificationUser>> FindByCriteriaAsync(List<Expression<Func<NotificationUser, bool>>> criteria, string orderBy,bool isDesc, CancellationToken cancellationToken = default);
        List<NotificationUser> FindByCriteria(List<Expression<Func<NotificationUser, bool>>> criteria, string orderBy, bool isDesc); 
        Task<PageList<NotificationUser>> FindByCriteriaAsync(List<Expression<Func<NotificationUser, bool>>> criteria, string orderBy, bool isDesc, int pageSize,int pageNumber, CancellationToken cancellationToken = default);
        PageList<NotificationUser> FindByCriteria(List<Expression<Func<NotificationUser, bool>>> criteria, string orderBy, bool isDesc, int pageSize, int pageNumber); 

Rename FInd to Get

File: server/src/TabTabGo.WebStream/TabTabGo.WebStream.SignalR/TabTabGo.WebStream.SignalR.csproj
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/TabTabGo.WebStream.SignalR/TabTabGo.WebStream.SignalR.csproj:1]: Still .net Standard2.0

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Repositories/EfNotificationRepository.cs
Comment (gmsamara) [inline server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage.EFCore/Repositories/EfNotificationRepository.cs:1]: Still use builder and not use the TabTabGo.Core framework

File: server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationRepository.cs
Comment (gmsamara) [reply server/src/TabTabGo.WebStream/NotificationStorage/TabTabGo.WebStream.NotificationStorage/Repository/INotificationRepository.cs:1]: still parameter name not changed

**Review by gmsamara — CHANGES_REQUESTED** [review]
Still need to update INotificationRepository . the other changes can be done later

**Review by gmsamara — APPROVED** [review]

**Pull request merged**